dockstore-webservice/src/main/java/io/dockstore/webservice/languages/LanguageHandlerInterface.java

```diff
@@ -325,6 +335,97 @@ default String getURLFromEntry(String dockerEntry, ToolDAO toolDAO) {
         return url;
     }
 
+    // TODO: Implement for DockerHub, then gitlab and seven bridges;
+    default Set<Image> getImagesFromRegistry(String toolsJSONTable) {
+        List<Map<String, String>> dockerTools = new ArrayList<>();
+        dockerTools = (ArrayList<Map<String, String>>)GSON.fromJson(toolsJSONTable, dockerTools.getClass());
+
+        // Eliminate duplicate docker strings
+        Set<String> dockerStrings = dockerTools.stream().map(dockertool -> dockertool.get("docker")).filter(Objects::nonNull).collect(Collectors.toSet());
+
+        Set<Image> dockerImages = new HashSet<>();
+        String errorKey = "error_message";
+        for (String image : dockerStrings) {
+            String[] parts = image.split("/");
+
+            Optional<String> response;
+            if (image.startsWith("quay.io/")) {
+                String[] splitDocker;
+                String[] splitTag;
+
+                try {
+                    splitDocker = image.split("/");
+                    splitTag = splitDocker[2].split(":");
+                } catch (ArrayIndexOutOfBoundsException ex) {
+                    LOG.error("URL to image on Quay incomplete", ex);
+                    break;
+                }
+
+
+                if (splitTag.length > 1) {
+                    String repo = splitDocker[1] + "/" + splitTag[0];
+                    String tagName = splitTag[1];
+                    response = getImageResponseFromQuay(image);
```

Comment:
1. You recalculate splitDocker and splitTag inside `getImageResponseFromQuay` when you already calculated them above on lines 357-358.
2. You're not going to used `repo` and `tagName` unless you get inside the `if (response.isPresent()...` statement below; I would move lines 366-367 to inside the if block, where you actually need/use them.

Reply:
not exactly what you asked for but did change